mcu/apollo3/CMakeLists.txt

```diff
@@ -0,0 +1,52 @@
+# Ambiq HAL
+#
+# Copyright (c) 2023 Ambiq Ltd <www.ambiq.com>
```

Comment:
Suggest using "Copyright (c) 2024, Ambiq Micro, Inc." to align with other source files. The same as other new added CMakelist.txt.